src/Compilers/CSharp/Test/Syntax/Parsing/StatementParsingTests.cs

```diff
@@ -4737,6 +4737,640 @@ void M()
             EOF();
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/67757")]
+        public void ParseSwitchStatementWithUnclosedRecursivePattern1()
+        {
+            UsingStatement("""
+                switch (obj)
+                {
+                    case Type { Prop: Type { }:
```

Comment:
Let's not address in this PR, but I am curious if a similar error recovery improvement is possible/worthwhile for

```cs
obj switch
{
    { Prop: { } => 1,
    { Prop: { } => 2
}
```

I wonder if the same TerminatorState flag could be reused as well.

Reply:
Converted to an issue